mmv1/api/resource.go

```diff
@@ -615,6 +621,28 @@ func (r Resource) RootProperties() []*Type {
 	return props
 }
 
+// Returns a sorted list of all "leaf" properties, meaning properties that have
+// no children.
+func (r Resource) LeafProperties() []*Type {
+	types := r.AllNestedProperties(google.Concat(r.RootProperties(), r.UserVirtualFields()))
+
+	// Remove types that have children, becasue we only want "leaf" fields
```

Comment:
sp - feel free to ignore

```suggestion
	// Remove types that have children, because we only want "leaf" fields
```